src/Campaigns/Migrations/Donations/AddCampaignId.php

```diff
@@ -86,7 +104,56 @@ public function run()
                     ->insert($donationMeta, ['%d', '%s', '%d']);
             }
         } catch (DatabaseQueryException $exception) {
-            throw new DatabaseMigrationException("An error occurred while adding campaign ID to the donation meta table", 0, $exception);
+            throw new DatabaseMigrationException("An error occurred while adding campaign ID to the donation meta table",
+                0, $exception);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getItemsCount(): int
+    {
+        return $this->query()->count();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBatchItemsAfter($lastId): ?array
+    {
+        $items = $this->query()
+            ->select('ID')
+            ->where('ID', $lastId, '>')
+            ->orderBy('ID')
+            ->limit($this->getBatchSize())
+            ->getAll();
+
+        if ( ! $items) {
+            return null;
         }
+
+        return [
+            min($items)->ID,
+            max($items)->ID,
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBatchSize(): int
+    {
+        return 100;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function hasIncomingData($lastProcessedId): ?bool
+    {
+        return $this->query()
+            ->where('ID', $lastProcessedId, '>')
+            ->count();
```

Comment:
Wouldn't this query everything after that ID? Shouldn't we apply `->limit(1)`, here?

Reply:
Yep, good catch!

Reply:
Resolved https://github.com/impress-org/givewp/pull/7755/commits/d844f0d1c3bdb8ab925950ea08db699a98ad54fd